clients/gitlabrepo/licenses.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitlabrepo
+
+// TODO:
+// add "github.com/xanzy/go-gitlab" to this list.
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+var errRepoUnsupported = errors.New("notsupported")
+
+type licensesHandler struct {
+	// TODO: glClient *gitlab.Client
+	once     *sync.Once
+	errSetup error
+	repourl  *repoURL
+	licenses []clients.License
+}
+
+func (handler *licensesHandler) init(repourl *repoURL) {
+	handler.repourl = repourl
+	handler.errSetup = nil
+	handler.once = new(sync.Once)
+}
+
+func (handler *licensesHandler) setup() error {
+	handler.once.Do(func() {
+		// TODO: find actual GitLab API, data type, and fields
```

Comment:
feel free to create an issue on the repo and tackle this in a follow-up PR.

Reply:
great - thanks, I'll do that for sure.

It was troublesome enough just to get this "dummy" component into the structure given that the build system was looking for it (that is for gitlab) and I could not figure out how to turn that off (in the build system). So now, if called, it will simply error as "not supported" -- but the legacy code will continue to do what it always did.

Reply:
Still TBD. resolving this comment now and will issue the PR once this code is merged.

Reply:
Still TBD. resolving this comment now and will issue the PR once this code is merged.

Reply:
Still TBD. resolving this comment now and will issue the PR once this code is merged.